react-form-controls/src/NewsletterForm.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { value: '' };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  render() {
+    return (
+      <form>
+        <label htmlFor="email">Email</label>
+        <input id="email" type="email" onChange={this.handleChange}></input>
```

Comment:
Be sure to include a `value` prop so that your input element is fully controlled by react